data_steward/cdr_cleaner/cleaning_rules/person_id_validator.py

```diff
@@ -51,10 +51,17 @@
     -- Need to use Jinja templates to make this decision --
     {% if dataset[0].isalpha() %}
     -- should expect extension tables --
-     and lower(moe.src_id) <> 'ppi/pm'
+     and NOT REGEXP_CONTAINS(lower(moe.src_id), r'(?i)(Portal)')
     {% else %}
     -- should expect mapping tables --
-     and lower(moe.src_hpo_id) <> 'rdr'
+     and lower(moe.src_hpo_id) NOT IN (
+        SELECT 
+            hpo_id
+        FROM
+            `{{project_id}}.pipeline_tables.site_maskings`
+        WHERE NOT
+            REGEXP_CONTAINS(src_id, r'(?i)(PPI/PM)|(EHR site)')
+     )
```

Comment:
You might write this differently to avoid the double negative confusion. Suggestions seen below.

```
and lower(moe.src_hpo_id) IN (
        SELECT 
            hpo_id
        FROM
            `{{project_id}}.pipeline_tables.site_maskings`
        WHERE
            REGEXP_CONTAINS(src_id, r'(?i)(EHR site)')
```

Below is another simplification. But either would be fine.

```
and  REGEXP_CONTAINS(moe.src_hpo_id, r'(?i)(EHR site)'
```

Reply:
Thanks for meeting with me Luis. I see I had a misunderstanding when making the suggestions. Your changes should work as expected I agree.